server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/LocalGroupByExecutor.java

```diff
@@ -150,12 +148,6 @@ private void calcFromBatch(BatchData batchData, long curStartTime, long curEndTi
         result.updateResultFromPageData(batchIterator, curStartTime, curEndTime);
       }
     }
-    lastReadCurArrayIndex = batchData.getReadCurArrayIndex();
-    lastReadCurListIndex = batchData.getReadCurListIndex();
-    // can calc for next interval
-    if (batchData.hasCurrent()) {
-      preCachedData = batchData;
-    }
```

Comment:
You delete these lines in `calcFromBatch` and then add them in the process of `readAndCalcFromPage`, but `calcFromBatch` is still called by `calcFromCacheData` where you forgot to change. So why you want move these lines outside of this method.

Reply:
fixed